src/modules/wallets/sagas.js

```diff
@@ -257,7 +263,8 @@ const withdraw = function * ({ blockchainKey, ...payload }) {
 
 export default function * () {
   yield all([
-    takeLatest(T.login.LOGGED, checkRegistrations),
+    fork(checkRegistrations),
+    fork(generatePrimaryWallet),
```

Comment:
Both checkRegistrations and generatePrimaryWallet check existence of primaryKey stored, So logic is done twice...

=> generatePrimaryWallet should be a consequence of checkRegistrations saying primary blockchain is not registered, not happening concurrently